node/src/reactor.rs

```diff
@@ -443,9 +464,43 @@ where
         self.event_count += 1;
     }
 
+    /// Get both the allocated and total memory from sys-info + jemalloc
+    fn get_allocated_memory() -> Option<(u64, u64)> {
```

Comment:
I'd rather see a small struct holding these two values rather than a tuple, especially since the two values are of the same type.